ElectronApp/src/__tests__/Nav.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { MemoryRouter } from 'react-router-dom';
+
+import '@testing-library/jest-dom';
+import App from '../renderer/App';
+import Nav from '../renderer/Nav';
+import DataEntry from '../renderer/routes/data-entry';
+import DataReports from '../renderer/routes/data-reports';
+import FieldForms from '../renderer/routes/field-forms';
+
+describe('Nav', () => {
+  const leftClick = { button: 0 };
+
+  it('should render', () => {
+    expect(render(<Nav />, { wrapper: MemoryRouter })).toBeTruthy();
+  });
+
+  it('should switch to Data Entry', () => {
+    render(<App />, { wrapper: MemoryRouter });
+
+    userEvent.click(screen.getByText(/data entry/i), leftClick);
+
+    expect(render(<DataEntry />, { wrapper: MemoryRouter })).toBeTruthy();
+  });
+
+  it('should switch to Data Reports', () => {
+    render(<App />, { wrapper: MemoryRouter });
+
+    userEvent.click(screen.getByText(/data reports/i), leftClick);
+
+    expect(render(<DataReports />, { wrapper: MemoryRouter })).toBeTruthy();
+  });
+
+  it('should switch to Field Forms', () => {
+    render(<App />, { wrapper: MemoryRouter });
+
+    userEvent.click(screen.getByText(/field forms/i), leftClick);
+
+    expect(render(<FieldForms />, { wrapper: MemoryRouter })).toBeTruthy();
```

Comment:
This line simply checks that we render the FieldForms component, and it will always return true (assuming we wrote FieldForms correctly), irrespective of the previous two statements!

To make our unit test independent of the implementation of FieldForms, try this instead:

```
// field-forms.tsx
<div data-testid="field-forms">
  ...
</div>
```
```
// Nav.test.tsx
it('should switch to Field Forms', () => {
  render(<App />, { wrapper: MemoryRouter });

  userEvent.click(screen.getByText(/field forms/i), leftClick);

  expect(screen.getByTestId("field-forms")).toBeTruthy();
});
```

Reply:
Gotcha! Thanks Justin